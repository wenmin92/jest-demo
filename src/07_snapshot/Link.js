import React, { useState } from "react";

const Link = ({ page, children }) => {
    const [status, setStatus] = useState('normal');
    const handleMouseEnter = () => {
        setStatus('hovered');
    }

    const handleMouseLeave = () => {
        setStatus('normal');
    }

    return (
        <a className={status}
            href={page || '#'}
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
        >
            {children}
        </a>
    );
};

export default Link;
