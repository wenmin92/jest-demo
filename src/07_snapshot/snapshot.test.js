import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Link from './Link';

// test('snapshot first view', () => {
//     document.body.innerHTML = '<div id="root"></div>';
//     ReactDOM.render(<Link page="http://www.facebook.com">Facebook</Link>, document.querySelector('#root'));
//     expect(document.body.innerHTML).toMatchSnapshot();
// });

// test('snapshot second view', () => {
//     expect('a string').toMatchSnapshot();
// });

// test('snapshot third view', () => {
//     expect({ num: 1, string: 'xyz', date: new Date() }).toMatchSnapshot({
//         date: expect.any(Date)
//     });
// })

// test('Link changes the class when hovered', () => {
//     const component = renderer.create(
//         <Link page="http://www.facebook.com">Facebook</Link>
//     );
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();

//     // // manually trigger the callback
//     // tree.props.onMouseEnter();
//     // // re-rendering
//     // tree = component.toJSON();
//     // expect(tree).toMatchSnapshot();

//     // // manually trigger the callback
//     // tree.props.onMouseLeave();
//     // // re-rendering
//     // tree = component.toJSON();
//     // expect(tree).toMatchSnapshot();
//     component.unmount();
// });

it('inline snapshot', () => {
    const obj = {
        msg: 'hello, jest!',
        date: new Date(),
    };
    expect(obj).toMatchInlineSnapshot({
        date: expect.any(Date)
    }, `
Object {
  "date": Any<Date>,
  "msg": "hello, jest!",
}
`);
});
