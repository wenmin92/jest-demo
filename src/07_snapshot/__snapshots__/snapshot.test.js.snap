// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="https://example.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Example Site
</a>
`;

exports[`snapshot first view 1`] = `"<div id=\\"root\\"><a class=\\"normal\\" href=\\"http://www.facebook.com\\">Facebook</a></div>"`;

exports[`snapshot second view 1`] = `"a string"`;

exports[`snapshot third view 1`] = `
Object {
  "date": Any<Date>,
  "num": 1,
  "string": "xyz",
}
`;
